compiler/rustc/src/main.rs

```diff
@@ -1,5 +1,7 @@
 // We need this feature as it changes `dylib` linking behavior and allows us to link to `rustc_driver`.
 #![feature(rustc_private)]
+// Several crates are depended upon but unused so that they are present in the sysroot
+#![expect(unused_crate_dependencies)]
```

Comment:
Some crates might need to be present in sysroot because they might be depended upon externally, right? Not sure if that's a problem.